### PR TITLE
Adding log prefixkey to html logger

### PR DESCRIPTION
**Pull request by hvinett**
## Description
Adding Log Prefix key for html logger

### CODE REVIEW

File: src/Microsoft.TestPlatform.Extensions.HtmlLogger/HtmlLogger.cs [hunk 1]
```diff
@@ -273,11 +274,21 @@ public void TestRunCompleteHandler(object sender, TestRunCompleteEventArgs e)
...
             };
-            var isLogFileNameParameterExists = parametersDictionary.TryGetValue(HtmlLoggerConstants.LogFileNameKey,
-                               out string logFileNameValue);
-            if (isLogFileNameParameterExists && !string.IsNullOrWhiteSpace(logFileNameValue))
+
+            if (this.parametersDictionary.TryGetValue(HtmlLoggerConstants.LogFilePrefixKey, out string logFilePrefixValue) && !string.IsNullOrWhiteSpace(logFilePrefixValue))
+            {
+
+                var framework = this.parametersDictionary[DefaultLoggerParameterNames.TargetFramework] ?? string.Empty;
+                framework = NuGetFramework.Parse(framework).GetShortFolderName();
```
Comment (singhsarab) [inline src/Microsoft.TestPlatform.Extensions.HtmlLogger/HtmlLogger.cs]: What happens when framework is null/empty

File: src/Microsoft.TestPlatform.Extensions.HtmlLogger/HtmlLogger.cs [hunk 2]
```diff
@@ -273,11 +274,21 @@ public void TestRunCompleteHandler(object sender, TestRunCompleteEventArgs e)
...
+                logFilePrefixValue = logFilePrefixValue.Replace(".html", string.Empty) + "_" + framework + DateTime.Now.ToString("_yyyyMMddHHmmss", DateTimeFormatInfo.InvariantInfo) + $".{HtmlLoggerConstants.HtmlFileExtension}";
```
Comment (singhsarab) [inline src/Microsoft.TestPlatform.Extensions.HtmlLogger/HtmlLogger.cs]: Change/remove replace

File: src/Microsoft.TestPlatform.Extensions.HtmlLogger/HtmlLogger.cs (hunk 1 above)
Comment (hvinett) [reply src/Microsoft.TestPlatform.Extensions.HtmlLogger/HtmlLogger.cs]: Throws exception if framework is empty. so proceeding with this step only when framework is not null

File: src/Microsoft.TestPlatform.Extensions.HtmlLogger/HtmlLogger.cs (hunk 2 above)
Comment (hvinett) [reply src/Microsoft.TestPlatform.Extensions.HtmlLogger/HtmlLogger.cs]: removed

File: src/Microsoft.TestPlatform.Extensions.HtmlLogger/HtmlLogger.cs [hunk 3]
```diff
@@ -137,6 +138,16 @@ public void Initialize(TestLoggerEvents events, Dictionary<string, string> param
...
 
             parametersDictionary = parameters;
+
+            if (parameters.TryGetValue(HtmlLoggerConstants.LogFilePrefixKey, out string logFilePrefixValue) && parameters.TryGetValue(HtmlLoggerConstants.LogFileNameKey, out string logFileNameValue))
+            {
+                var htmlParameterErrorMsg = string.Format(CultureInfo.CurrentCulture,
+                        HtmlResource.PrefixAndNameProvidedError);
+
+                EqtTrace.Error(htmlParameterErrorMsg);
+                 throw new ArgumentException(htmlParameterErrorMsg);
```
Comment (singhsarab) [inline src/Microsoft.TestPlatform.Extensions.HtmlLogger/HtmlLogger.cs]: formatting ?

File: src/Microsoft.TestPlatform.Extensions.HtmlLogger/HtmlLogger.cs [hunk 4]
```diff
@@ -273,11 +284,26 @@ public void TestRunCompleteHandler(object sender, TestRunCompleteEventArgs e)
                 PassPercentage = (PassedTests * 100) / TotalTests,
                 TotalRunTime = GetFormattedDurationString(e.ElapsedTimeInRunningTests),
             };
-            var isLogFileNameParameterExists = parametersDictionary.TryGetValue(HtmlLoggerConstants.LogFileNameKey,
-                               out string logFileNameValue);
-            if (isLogFileNameParameterExists && !string.IsNullOrWhiteSpace(logFileNameValue))
+
+            if (this.parametersDictionary.TryGetValue(HtmlLoggerConstants.LogFilePrefixKey, out string logFilePrefixValue) && !string.IsNullOrWhiteSpace(logFilePrefixValue))
+            {
```
Comment (singhsarab) [inline src/Microsoft.TestPlatform.Extensions.HtmlLogger/HtmlLogger.cs]: Remove the extra line

File: test/Microsoft.TestPlatform.Extensions.TrxLogger.UnitTests/TrxLoggerTests.cs [hunk 5]
```diff
@@ -712,7 +712,6 @@ private void ValidateDateTimeInTrx(string trxFileName)
 
         [TestMethod]
         [DataRow("results")]
-        [DataRow("results.trx")]
```
Comment (singhsarab) [inline test/Microsoft.TestPlatform.Extensions.TrxLogger.UnitTests/TrxLoggerTests.cs]: instead add separate test, use data row with 2 arguments, one is the input and other expected output

Comment (hvinett) [reply test/Microsoft.TestPlatform.Extensions.TrxLogger.UnitTests/TrxLoggerTests.cs]: '.' is now like other alphabets right? do we really need another test case?

File: src/Microsoft.TestPlatform.Extensions.HtmlLogger/HtmlLogger.cs (hunk 4 above)
Comment (hvinett) [reply src/Microsoft.TestPlatform.Extensions.HtmlLogger/HtmlLogger.cs]: removed

File: src/Microsoft.TestPlatform.Extensions.TrxLogger/TrxLogger.cs [hunk 6]
```diff
@@ -489,10 +489,13 @@ private void DeriveTrxFilePath()
             {
                 if (!string.IsNullOrWhiteSpace(logFilePrefixValue))              
                 {
-                    var framework = this.parametersDictionary[DefaultLoggerParameterNames.TargetFramework] ?? string.Empty;
-                    framework = NuGetFramework.Parse(framework).GetShortFolderName();
+                    var framework = this.parametersDictionary[DefaultLoggerParameterNames.TargetFramework];
+                    if (framework != null)
+                    {
+                        framework = NuGetFramework.Parse(framework).GetShortFolderName();
```
Comment (vagisha-nidhi) [inline src/Microsoft.TestPlatform.Extensions.TrxLogger/TrxLogger.cs]: logFilePrefixValue = logFilePrefixValue + "_" + framework only when framework value is not null

File: src/Microsoft.TestPlatform.Extensions.TrxLogger/TrxLogger.cs [hunk 7]
```diff
@@ -489,10 +489,13 @@ private void DeriveTrxFilePath()
...
+                    }
 
-                    logFilePrefixValue = logFilePrefixValue.Replace(".trx", string.Empty) + "_" + framework + DateTime.Now.ToString("_yyyyMMddHHmmss", DateTimeFormatInfo.InvariantInfo) + this.trxFileExtension;
+                    logFilePrefixValue = logFilePrefixValue + "_" + framework + DateTime.Now.ToString("_yyyyMMddHHmmss", DateTimeFormatInfo.InvariantInfo) + this.trxFileExtension;
```
Comment (vagisha-nidhi) [inline src/Microsoft.TestPlatform.Extensions.TrxLogger/TrxLogger.cs]: logFilePrefixValue = logFilePrefixValue + "_" + framework only when framework value is not null

**Review by vagisha-nidhi — APPROVED** [review]

**Pull request merged**